### PR TITLE
Add dagre@0.7.4 w/ npm auto-update

### LINKED ISSUE #11102: [Request] Add dagre
**Library name:** dagre
**Git repository url:** https://github.com/cpettitt/dagre
**npm package name or url** (if there is one): https://www.npmjs.com/package/dagre
**License (List them all if it's multiple):** MIT
**Official homepage:** 
**Wanna say something? Leave message here:**

### PR DESCRIPTION
**Pull request by dakshshah96**
close #11102, cc @cpettitt

Pull request for issue: #11102

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/cpettitt/dagre
 * Official website (optional, not the repository): https://github.com/cpettitt/dagre/wiki
 * NPM package url (optional): https://www.npmjs.com/package/dagre
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 89
   - Count of stars: 1,245
   - Count of forks: 176
 * NPM download stats (optional):
   - Downloads in the last day: 617
   - Downloads in the last week: 4,167
   - Downloads in the last month: 18,886

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/dagre/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "filename": "dagre.min.js",
+  "description": "Graph layout for JavaScript",
+  "homepage": "https://github.com/cpettitt/dagre/wiki/",
+  "keywords": [
+    "graph",
+    "layout"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/cpettitt/dagre"
```
Comment (sufuf3) [inline ajax/libs/dagre/package.json]: Maybe we can follow the format as the same as upstream. ref: https://github.com/cpettitt/dagre/blob/master/package.json#L35-L38
`"url": "https://github.com/cpettitt/dagre.git"`

File: ajax/libs/dagre/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "license": "MIT",
+  "author": "Chris Pettitt <@cpettitt>",
```
Comment (sufuf3) [inline ajax/libs/dagre/package.json]: Maybe we can follow the format as the same as upstream. ref: https://github.com/cpettitt/dagre/blob/master/package.json#L5
`"author": "Chris Pettitt <cpettitt@gmail.com>"`

**Comment by dakshshah96** [comment]
@hare1039 @extend1994 v0.0.1~v0.5.4 of the library does not have compiled files which can be directly used like `dagre.js`, `dagre.core.js`, `dagre.min.js`, `dagre.core.min.js`. 

v0.0.1~v0.5.4 consist only of files which are needed to build the library. 

What should be done in this case?

**Comment by extend1994** [comment]
@dakshshah96 You can open an issue to ask the author(s) or the maintainer(s) about what should we add in v0.0.1~v0.5.4. He/She/They might tell(s) you that 
1. We don't need to host those versions because they are not packaged by the author(s)
2. Where are the files we need.
3. Wait the author(s) or the maintainer(s) for providing packaged files.
Here is a example of the issue: https://github.com/quark-dev/Phonon-Framework/issues/219

**Comment by dakshshah96** [comment]
@extend1994 Communicated with the author at [issue #214 on dagre](https://github.com/cpettitt/dagre/issues/214).

**Comment by PeterDaveHello** [comment]
@sufuf3 didn't this need to be rebased with the latest master branch?

**Comment by dakshshah96** [comment]
@extend1994 @PeterDaveHello @sufuf3 Author recommends not adding the old versions since it has already been a couple of years since even the latest version was released.

**Comment by PeterDaveHello** [comment]
@dakshshah96 thanks

**Comment by sufuf3** [comment]
@dakshshah96 Thank you.
@PeterDaveHello Sorry for no explaining. My review part was modified, and as @dakshshah96 said, the author suggested ignoring those versions, because they had a different API and the current release has been out for a couple of years now - crazy to realize it has been that long!
Now, I think this PR looks good to me.

File: ajax/libs/dagre/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "dagre",
+  "version": "0.7.4",
+  "filename": "dagre.min.js",
+  "description": "Graph layout for JavaScript",
+  "homepage": "https://github.com/cpettitt/dagre/wiki/",
```
Comment (pvnr0082t) [inline ajax/libs/dagre/package.json]: We'll use git repo as homepage by default if the custom homepage is not found, so there is no need to set homepage to its git repo. Let's remove it. Thanks.

Comment (dakshshah96) [inline ajax/libs/dagre/package.json]: @pvnr0082t This is the link to the wiki including examples and configuration options which isn't included in the git repo main page. Seems better for user to keep it, no?

Comment (pvnr0082t) [inline ajax/libs/dagre/package.json]: Okay, let's keep it.

**Pull request merged**